src/main/java/database/DatabaseController.java

```diff
@@ -116,6 +156,47 @@ public void createUser(String username, String password, String email) {
         }
     }
 
+    /**
+     * removes listing from csv file based on id given
+     * writes to a new file without listing to be removed, and then renamed
+     * helper for checkoutRemoveListings
+     * also used for when user wants to take down a listing
+     *
+     * @param ID id passed to removeListing
+     */
+    public void removeListing(int ID) throws IOException {
+        try {
+            FileWriter listingsWriter = new FileWriter("../entities/Listings.csv");
+            File listings = new File("../entities/Listings.csv");
+            File temp = File.createTempFile("temp", ".csv", new File("../entities/"));
+
+            BufferedReader reader = new BufferedReader(new FileReader(String.valueOf(listingsWriter)));
+            CSVWriter writer = new CSVWriter(new FileWriter(temp));
+
+            String currLine;
+
+            while ((currLine = reader.readLine()) != null) {
+                String[] listing = currLine.split(";");
+                int listingID = Integer.parseInt(listing[0]);
+                if (listingID == ID) {
+                    continue;
+                }
+                writer.writeNext(listing);
```

Comment:
what's the point of this? are you going to replace the continue with code where you actually delete the line?

Reply:
also, we will have to handle deleting the listing from all user objects too, since the csv file will not automatically do it for us :(

Reply:
Delete the listing from all User object's carts? Is it bad to just reset the cart whenever they login?